scripts/make-types.ts

```diff
@@ -0,0 +1,52 @@
+// Run as `node -r sucrase/register scripts/make-types.ts`
+
+import { makeNodeDisklet } from 'disklet'
+import prettier from 'prettier'
+
+function tsToFlow(code: string): string {
+  // First, use Prettier to add semicolons everywhere:
+  const formatted = prettier.format(code, {
+    parser: 'typescript',
+    semi: true
+  })
+
+  const fixed = formatted
+    // Language differences:
+    .replace(/\breadonly /g, '+')
+    .replace(/\bexport declare const\b/g, 'declare export var')
+    .replace(/\bexport declare\b/g, 'declare export')
+    .replace(/\binterface (\w+) {/g, 'type $1 = {')
+    .replace(/\binterface (\w+) extends (\w+) {/g, 'type $1 = { ...$2,')
+    .replace(/\bunknown\b/g, 'mixed')
+    .replace(/\| undefined\b/g, '| void')
+    .replace(/: undefined\b/g, ': void')
+
+    // Builtin types:
+    .replace(/\b(\w+): ComponentType</g, '$1: React$ComponentType<')
+    .replace(/\bPartial<(\w+)>/g, '$Rest<$1, { ... }>')
+
+  return '// @flow\n\n' + fixed
+}
```

Comment:
Do we need to put back the shims?

Reply:
The "Builtin types" section contains the only shims we cared about.